weave-js/src/components/Panel2/PanelObject.tsx

```diff
@@ -369,18 +370,20 @@ const PanelObjectChild: React.FC<
           <div style={{paddingLeft: '1em', height: 400, width: '100%'}}>
             {defaultLanguageBinding.printType(childType, true)}
             {/* TODO: This is not probably what we always want! - came from weaveflow */}
-            <ChildPanel
-              config={config?.children?.[k] ?? childNode}
-              updateConfig={newConfig => {
-                updateConfig({
-                  ...config,
-                  children: {
-                    ...config?.children,
-                    [k]: newConfig,
-                  },
-                });
-              }}
-            />
+            <PanelInteractContextProvider>
```

Comment:
While this fixes the specific bug you are investigating, it leaves open the problem that anywhere a `ChildPanel` is rendered in the workspace, we will crash. I would ask myself the following questions:
1. What is `PanelInteractContext`? Looking at its code and implementation, it allows child panels to do things like open a sidebar configuration for themselves for example. It is the root "interaction" context. 
1. When is `PanelInteractContextProvider` needed? - Basically all the panels that import `PanelInteractContext`. But the most popular user is `ChildPanel`.
2. Under what conditions is `PanelInteractContext` provided? Basically search for `<panelinteractContext` and you will see that it is provided at the top-level of most weave entrypoints (pagePanel, entrypoint, dashboard, etc..).
3. Question 1: Why is `panelinteractContext` not provided at the root of workspaces? The answer here is: PanelInteractContext was originally just used in boards... (notice how the workspace doesn't have a global sidebar configuration space for example). In general, it is part of the newer weave-board style panel architecture, which is not fully adopted in workspaces. 
4. What changed? Well inferred by the comment above `{/* TODO: This is not probably what we always want! - came from weaveflow */}`, this was something where we let the new weave stuff bleed into the old weave stuff. 

There are other smells here: Why is it that just `PanelObject` needs this context provider? Is there something special about panel object? Do we need to wrap this around every ChildPanel? What happens if multiple panel objects are nested? 

I think if we note that `PanelInteractContext` is intended to be a page-level global object that maintains state & config interactions across panels, then it would make more sense to either put the `<PanelInteractContextProvider>` wrapping the entire workspace - similar to what is done for reports.

The drawback to this (and the current approach) is that calls like `useSetInteractingPanel` will not have an effect since there is no delegate to open a config drawer for example.


---

To be explicit here: `PanelInteractContextProvider` should not go in `PanelObject`. Instead, it should be much higher up the react tree, likely around the root of the workspace.

Reply:
@tssweeney thanks for the detailed & insightful comment. Agreed that the current fix is specific and doesn't address the broader issue with ChildPanel and PanelInteractContext. Considering that PanelInteractContext should be at the root-level, I put it inside the WeaveFeaturesContextProvider which is rendered at the root-level for all instances of Weave panels in the app.